lib/logstash/util/accessors.rb

```diff
@@ -48,7 +48,9 @@ def strict_set(accessor, value)
 
     def del(accessor)
       target, key = lookup(accessor)
-      target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
+      unless target.nil?
```

Comment:
I think in this case we want to actually skip if it's nil only, not false. If we hav ea field that has a false value, removing it should still work. It would be weird if you tried to delete a field with a boolean false value and the delete didn't work.